### PR TITLE
feature: use total visit time

### PR DESCRIPTION
**Pull request by Kovalenko77**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -88,23 +88,43 @@ return {
...
+    if (seconds < 10) {
         seconds = '0' + seconds;
     }
     document.getElementById('timer').innerHTML = `${hour} : ${minute} : ${seconds}`;
+    saveTotalSecondsValue(time);
+}
+
+function saveTotalSecondsValue(time) {
+    localStorage.setItem('totalSeconds',time);
+    // console.log('restoredTime',restoredTime)
```
Comment (Kovalenko77) [inline index.js]: why do we need this comment?

Comment (Kovalenko77) [reply index.js]: removed. Thanks.

File: index.js [hunk 2]
```diff
@@ -88,23 +88,43 @@ return {
...
-    let minute = Math.floor((totalSeconds - hour * 3600) / 60);
-    let seconds = totalSeconds - (hour * 3600 + minute * 60);
-    if(hour < 10) {
+let restoredTime = localStorage.getItem('totalSeconds');
+
+const timer = setInterval(a, 1000);
+function a() {
+    let arg;
+    if (restoredTime) {
+        arg = localStorage.getItem('totalSeconds');
```
Comment (Kovalenko77) [inline index.js]: duplication

File: index.js [hunk 3]
```diff
@@ -42,7 +42,13 @@ function init() {
     const restoredNotificationsCheckboxValue = localStorage.getItem('notifications');
     formCheckboxElem.checked = JSON.parse(restoredNotificationsCheckboxValue);
     const restoredLang = localStorage.getItem('select-language');
-    changeLanguage(restoredLang);
+    let arg;
+    if (restoredLang) {
+        arg = restoredLang;
+    } else {
+        arg = 'en';
```
Comment (romanalesenkov1) [inline index.js]: 'en' should be moved to constant

File: index.js [hunk 4]
```diff
@@ -42,7 +42,13 @@ function init() {
     const restoredNotificationsCheckboxValue = localStorage.getItem('notifications');
     formCheckboxElem.checked = JSON.parse(restoredNotificationsCheckboxValue);
     const restoredLang = localStorage.getItem('select-language');
-    changeLanguage(restoredLang);
+    let arg;
+    if (restoredLang) {
+        arg = restoredLang;
```
Comment (romanalesenkov1) [inline index.js]: what is arg? it is a bad name for a variable

File: index.js [hunk 5]
```diff
@@ -42,7 +42,13 @@ function init() {
...
+    } else {
+        arg = 'en';
+    }
+    changeLanguage(arg);
```
Comment (romanalesenkov1) [inline index.js]: why don't you just use 
```suggestion
    changeLanguage(restoredLang || 'en');
```

File: index.js [hunk 6]
```diff
@@ -55,6 +61,9 @@ function translateContent(lang) {
     const labelEls = document.querySelectorAll('[translate]');
     for (let i = 0; i < labelEls.length; i++ )  { 
         const translationKey = labelEls[i].getAttribute('translate');
+        console.log('LANGUAGE_LABEL_CONTENT[lang]',LANGUAGE_LABEL_CONTENT[lang]);
```
Comment (romanalesenkov1) [inline index.js]: unneeded console.logs

File: index.js [hunk 7]
```diff
@@ -87,24 +96,40 @@ return {
...
     }
-    if(seconds < 10) {
+    if (seconds < 10) {
         seconds = '0' + seconds;
     }
     document.getElementById('timer').innerHTML = `${hour} : ${minute} : ${seconds}`;
+    saveTotalSecondsValue(time);
+}
+
+function saveTotalSecondsValue(time) {
```
Comment (romanalesenkov1) [inline index.js:108]: why it is global public function?

move all timer functionality to store. And hide all private functions within IIFE

File: index.js [hunk 8]
```diff
@@ -87,24 +96,40 @@ return {
...
+    localStorage.setItem('totalSeconds',time);
```
Comment (romanalesenkov1) [inline index.js]: add a space after a comma

File: index.js [hunk 9]
```diff
@@ -87,24 +96,40 @@ return {
...
+    countTimer(arg);
+}
+
+function countTimer(time) {
+    ++time;
+    let hour = Math.floor(time / 3600);
+    let minute = Math.floor((time - hour * 3600) / 60);
+    let seconds = time - (hour * 3600 + minute * 60);
+    if (hour < 10) {
         hour = '0' + hour;
```
Comment (romanalesenkov1) [inline index.js]: there is a native JS function that does that. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

File: index.js (hunk 8 above)
Comment (Kovalenko77) [reply index.js]: Removed. Thsnks.

File: index.js (hunk 9 above)
Comment (Kovalenko77) [reply index.js]: Helpful advice! Thanks.

File: index.js (hunk 7 above)
Comment (Kovalenko77) [reply index.js:108]: Done. Thanks.

File: index.js (hunk 6 above)
Comment (Kovalenko77) [reply index.js]: Removed. Thanks.

File: index.js (hunk 5 above)
Comment (Kovalenko77) [reply index.js]: Corrected. Thanks.

File: index.js (hunk 3 above)
Comment (Kovalenko77) [reply index.js]: Done. Thanks.

File: index.js (hunk 4 above)
Comment (Kovalenko77) [reply index.js]: Corrected. Thanks.

File: index.js (hunk 2 above)
Comment (Kovalenko77) [reply index.js]: Done. Thanks.

File: index.js [hunk 10]
```diff
@@ -42,7 +42,10 @@ function init() {
     const restoredNotificationsCheckboxValue = localStorage.getItem('notifications');
     formCheckboxElem.checked = JSON.parse(restoredNotificationsCheckboxValue);
     const restoredLang = localStorage.getItem('select-language');
-    changeLanguage(restoredLang);
+    const defaultLanguage = 'en';
```
Comment (romanalesenkov1) [inline index.js]: please, use UPPER_CASE_SNAKE for these kind of constants

Comment (romanalesenkov1) [inline index.js]: it is better to move this const outside this function, so it'll never be called more then one time

File: index.js [hunk 11]
```diff
@@ -83,28 +86,33 @@ function onChangeLanguage(event) {
...
-    if(minute < 10) {
-        minute = '0' + minute;
-    }
-    if(seconds < 10) {
-        seconds = '0' + seconds;
-    }
-    document.getElementById('timer').innerHTML = `${hour} : ${minute} : ${seconds}`;
-}
\ No newline at end of file
+
```
Comment (romanalesenkov1) [inline index.js:117]: remove empty line.

You can leave 1 or 0 empty lines at the end of the file.

File: index.js (hunk 10 above)
Comment (Kovalenko77) [reply index.js]: Fixed. Thanks.

Comment (Kovalenko77) [reply index.js]: Fixed. Thanks.

File: index.js (hunk 11 above)
Comment (Kovalenko77) [reply index.js:117]: Removed. Thenks.

File: index.js (hunk 10 above)
Comment (romanalesenkov1) [reply index.js]: No. You didn't

Comment (Kovalenko77) [reply index.js]: Sorry. Already done. Thanks.

**Pull request merged**